test/QuarkBuilder.t.sol

```diff
@@ -641,6 +643,42 @@ contract QuarkBuilderTest is Test {
         assertNotEq(result.eip712Data.hashStruct, hex"", "non-empty hashStruct");
     }
 
+    function testBridgeTransferMaxFundUnavailableError() public {
+        QuarkBuilder builder = new QuarkBuilder();
+        PaymentInfo.PaymentMaxCost[] memory maxCosts = new PaymentInfo.PaymentMaxCost[](3);
+        maxCosts[0] = PaymentInfo.PaymentMaxCost({chainId: 1, amount: 0.5e6});
+        maxCosts[1] = PaymentInfo.PaymentMaxCost({chainId: 8453, amount: 0.1e6});
+        maxCosts[2] = PaymentInfo.PaymentMaxCost({chainId: 7777, amount: 0.1e6}); // Random L2 with no bridge support
+        Accounts.ChainAccounts[] memory chainAccountsList = new Accounts.ChainAccounts[](3);
+        chainAccountsList[0] = Accounts.ChainAccounts({
+            chainId: 1,
+            quarkStates: quarkStates_(address(0xa11ce), 12),
+            assetPositionsList: assetPositionsList_(1, address(0xa11ce), 8e6)
+        });
+        chainAccountsList[1] = Accounts.ChainAccounts({
+            chainId: 8453,
+            quarkStates: quarkStates_(address(0xb0b), 2),
+            assetPositionsList: assetPositionsList_(8453, address(0xb0b), 4e6)
+        });
+        chainAccountsList[2] = Accounts.ChainAccounts({
+            chainId: 7777,
+            quarkStates: quarkStates_(address(0xfe11a), 2),
+            assetPositionsList: assetPositionsList_(7777, address(0xfe11a), 5e6)
+        });
+
+        // User has total holding of 17 USDC, but only 12 USDC is available for transfer/bridge to 8453, and missing 5 USDC stuck in random L2 so will revert with FundsUnavailable error
+        // Actual number re-adjusted with max cost gas fee so that the:
+        // Request amount = 17 USDC - 0.5 USDC (max cost on main) - 0.1 USDC (max cost on Base) - 0.1 USDC (max cost on RandomL2) = 16.3 USDC
```

Comment:
Shouldn't this amount be `16.3 - 4 = 12.3` USDC? 4 USDC is already on chain 8453, meaning only 12.3 need to be bridged over.

Also, I think the math/formula might be wrong, as it double subtracts the max cost on the destination chain. e.g. 0.1 USDC will be subtracted twice. Once during the `totalAvailableAsset` calculation, and a second when subtracting the `balanceOnDstChain`.

Reply:
Hmm oh yeah, this is a bit confusing. 
But as this test case will actually get caught by asserFundsAvailable() way before the bridge actions take in place,  I'm still trying to see if I can find a way to "pass" `assertFundsAvailable()` but fail at bridge, since `assertFundsAvailable()` verify the `canBridge()` also. 

If based on the `assertFundsAvailable()` logics, then it is 16.3. 
As the request amount is the origin maxbalance, which is sum of all 3 chains - cost for each chain = 16.4

I think if we pass the `assertFundsAvailable()` and fail in the bridge actions, then yeah it should be 12.3.  
But I need to find the way to pass `assertFundsAvailable()` for that case. lol 😅 
-----------------------------

Re about forumula on `totalAvailableAsset` 
I'm trying to list out the calculation here,  I think I didn't see where it double substracted in balanceOnDstChain 😅 
(Maybe I'm missing something obvious here and probably need another 👀 to double check) 
```
User has A USDC on main, B USDC on Base, C USDC on Blast
Cost is a, b, c on main, base, blast. 
Transfermax on main.

So in bridge blocks variables looks like ... 

transferIntent.Amount = totalAvailableAsset = A + B + C - a- b - c
balanceOnSDstChain = Accounts.getBalanceOnChain(transferIntent.assetSymbol, transferIntent.chainId, chainAccountsList) = A

amountLeftToBridge = (A+B+C-a-b-c) - A = B+C-a-b-c
# Need to add `a` back to amountLeftToBridge to account for dst chain gas 
if (paymentToken is tranfer):
   amountLeftToBridge = B+C-a-b-c + a = B+C-b-c

```

Reply:
Part of me feel like `assertFundsAvailable()` will catch most errors that bridge actions will run into. 
If assertFundsAvailable() pass,  I think it might be safe to assume the bridge actions will pass successfully. 🤔

Reply:
Let me think about it a bit more, but I guess if we can't think of a test case that hits this error, that really means some of the code is unnecessary and should be removed right?

Reply:
Thinking about this again, I think the test case is fine actually. In my bridge refactor PR (#30), I switch out the untested `FundsAvailable` error for a more specific `NotEnoughFundsToBridge`, which I test there.

Reply:
Interesting, yeah #30 could reach that when there is different flow to bridge payment token. 

Yeah I think the assertFundsAvialable() only catches only on transfer token and when transfer token is also payment token.